Use SSL in dev and remove clutter

roesslerj opened the pull request:
Use SSL in dev and remove clutter (such as obsolete code and spring profiles). Improve documentation.

roesslerj committed:
Make callback pass

roesslerj committed:
Make the callback service work with the self-signed EFGS certificate

roesslerj committed:
Work on make upload use SSL for EFGS

roesslerj committed:
Make `docker-compose up` execute successful

roesslerj committed:
Create p12 with same password as everything else and remove no longer needed profiles and code

roesslerj committed:
Remove obsolete Provider classes

roesslerj committed:
Remove all mentions of the obsolete SSL profiles and remove all clutter like providers and stuff

roesslerj committed:
Add separate test with certificates

roesslerj committed:
Fix the name of the config param (from key-store-pass to key-store-password)

roesslerj committed:
Make download tests pass

roesslerj committed:
Make upload pass

roesslerj committed:
Fix markdownlint error

roesslerj committed:
Fix the name of the config param (from key-store-pass to key-store-password) ... continued

roesslerj committed:
Add now necessary SSL config and FEDERATION_GATEWAY_KEYSTORE_PATH to docker-compose

roesslerj committed:
Remove unnecessary code from commit "Make the callback service work with the self-signed EFGS certificate"

roesslerj committed:
Remove redundant `ssl.p12` and `contains_efgs_truststore.jks` files and references

roesslerj committed:
Fix erroneous checkin from "Make callback pass"

roesslerj committed:
Overlooked file when doing "Remove redundant `ssl.p12` and `contains_efgs_truststore.jks` files and references"

roesslerj committed:
Merge branch 'main' into feature/ssl-in-dev

ioangut commented on common/federation/src/main/java/app/coronawarn/server/common/federation/client/CloudFederationFeignHttpClientProvider.java:
```diff
@@ -34,45 +32,44 @@
-    return new DefaultApacheHttpClientFactory(HttpClientBuilder.create()
-        .setMaxConnPerRoute(connectionPoolSize)
-        .setMaxConnTotal(connectionPoolSize)
+    return new DefaultApacheHttpClientFactory(HttpClientBuilder.create() //
```
Would you like to add some comments here? If not, can these be removed?

roesslerj replied:
These line-end comments force formatting. Can be removed if you want...?

ioangut commented on services/submission/src/main/resources/application.yaml:
```diff
@@ -47,8 +47,8 @@ services:
         key-password: ${SSL_SUBMISSION_KEYSTORE_PASSWORD}
         key-store: ${SSL_SUBMISSION_KEYSTORE_PATH}
         key-store-password: ${SSL_SUBMISSION_KEYSTORE_PASSWORD}
-        trust-store: ${SSL_VERIFICATION_TRUSTSTORE_PATH}
```
I think this should not be changed in submission service. The verification truststore path should be different from federation.

roesslerj replied:
Oh, thanks for the catch!

ioangut commented on services/submission/src/main/resources/application.yaml:
```diff
@@ -47,8 +47,8 @@ services:
         key-store: ${SSL_SUBMISSION_KEYSTORE_PATH}
         key-store-password: ${SSL_SUBMISSION_KEYSTORE_PASSWORD}
-        trust-store: ${SSL_VERIFICATION_TRUSTSTORE_PATH}
-        trust-store-password: ${SSL_VERIFICATION_TRUSTSTORE_PASSWORD}
```
Same as above

hilmarf committed:
Merge branch 'main' into feature/ssl-in-dev

hilmarf commented on services/submission/src/main/java/app/coronawarn/server/services/submission/verification/VerificationServerClientConfiguration.java:
```diff
@@ -9,9 +8,9 @@
 public class VerificationServerClientConfiguration {
 
-  private final FeignClientProvider feignClientProvider;
+  private final CloudFeignClientProvider feignClientProvider;
```
What's the difference between `CloudFeignClientProvider` and `FeignClientProvider`?

roesslerj replied:
`FeignClientProvider` was an abstract interface that was implemented twice, with/without SSL, where `CloudFeignClientProvider` was the SSL version. Since this in now the only remaining version, we use it directly instead of the now obsolete interface.

roesslerj committed:
Remove line-end comments that enforce formatting for some IDEs

roesslerj commented on common/federation/src/main/java/app/coronawarn/server/common/federation/client/CloudFederationFeignHttpClientProvider.java:
```diff
@@ -34,45 +32,44 @@
-          .loadKeyMaterial(keyStorePath, keyStorePass.toCharArray(), keyStorePass.toCharArray())
-          .loadTrustMaterial(this.trustStore, this.trustStorePassword.toCharArray())
+      return SSLContextBuilder.create().loadKeyMaterial(keyStorePath,
+              (keyStorePass != null ? keyStorePass.toCharArray() : new char[] {}),
+              (keyStorePass != null ? keyStorePass.toCharArray() : new char[] {}))
+          .loadTrustMaterial(this.trustStore,
+              (this.trustStorePassword != null ? this.trustStorePassword.toCharArray() : new char[] {}))
```
Sure thing

roesslerj commented on docker-compose-test-secrets/readme.md:
```diff
@@ -2,3 +2,20 @@ About the secrets contained here for development and testing
+CWA-Backend (in prod maintained by TSI)
+---------------------------------------
+
+- `private.pem`: The secret used to generate all key material on CWA-Backend side.
```
Hmm. I thing the original wording was better, as other keys are derived from that key material, not only signed by it.

roesslerj commented on docker-compose-test-secrets/readme.md:
```diff
@@ -2,3 +2,20 @@ About the secrets contained here for development and testing
+EFGS (in prod maintained by EFGS, TSI receives only public keys)
+----------------------------------------------------------------
+
+- `efgs_signing_key.pem`: The secret used to generate all key material on EFGS side.
```
Same here. But I have no strong opinion, so can change if you want.

roesslerj committed:
Use `emptyCharrArrayIfNull` instead of three times same clunky if-construct

roesslerj committed:
Undo unnecessary changes to formatting...

roesslerj committed:
Catch of @ioangut in review: should not be changed in submission service. The verification truststore path should be different from federation.

roesslerj committed:
Fixing checkstyle violation:

Indentation: '"app.coronawarn.server.services.callback"' has incorrect indentation level 2, expected level should be 4.

roesslerj committed:
Belongs to change Catch of @ioangut in review: should not be changed in submission service. The verification truststore path should be different from federation.
